controllers/discovery/searchUser.js

```diff
@@ -11,77 +11,92 @@ const {sequelize} = require('../../databases/models');
-            GROUP BY 
-                "User"."user_id"
-            ORDER BY  
-                "karma_score" DESC,
-                "user_id_follower" ASC,
-                "followersCount" DESC
-            LIMIT :limit`,
+          u.user_id,
+          u.human_id,
+          u.country_code,
+          u.username,
+          u.real_name,
+          u.created_at,
+          u.updated_at,
+          u.last_active_at,
+          u.status,
+          u.profile_pic_path,
+          u.bio,
+          u.is_banned,
+          u.is_anonymous,
+          u.allow_anon_dm,
+          u.only_received_dm_from_user_following,
+          u.is_backdoor_user,
+          u.blocked_by_admin,
+          u.verified_status,
+          u.combined_user_score,
+          u.karma_score,
+          u.is_karma_unlocked,
+          (SELECT COUNT(*) FROM user_follow_user WHERE user_id_followed = u.user_id) AS followersCount,
+          ARRAY(
+              SELECT name
+              FROM (
+                  SELECT name, ROW_NUMBER() OVER (ORDER BY tp.topic_id) AS row_num
+                  FROM topics AS tp
+                  LEFT JOIN user_topics AS utp ON tp.topic_id = utp.topic_id
+                  WHERE utp.user_id = u.user_id AND tp.topic_id IN (
+                      SELECT tp.topic_id 
+                      FROM topics AS tp
+                      LEFT JOIN user_topics AS utp ON tp.topic_id = utp.topic_id
+                      WHERE utp.user_id = :userId
+                  )
+              ) AS subquery
+              WHERE row_num <= 3
+          ) AS community_info,
+          EXISTS (
+              SELECT 1
+              FROM user_follow_user AS f2
+              WHERE f2.user_id_follower = :userId
+              AND f2.user_id_followed = u.user_id
+          ) AS is_followed,
+          (SELECT user_id_follower FROM user_follow_user WHERE user_id_followed = u.user_id AND user_id_follower = :userId) AS user_id_follower
+      FROM 
+          users AS u
+      WHERE 
+          (u.username ILIKE :likeQuery
+          AND u.user_id != :userId
+          AND u.is_anonymous = false
+          AND u.is_banned = false 
+          AND u.verified_status != 'UNVERIFIED') 
+      ORDER BY  
+          u.karma_score DESC,
+          is_followed ASC,
+          followersCount DESC
+      LIMIT :limit`,
       {
         type: QueryTypes.SELECT,
         replacements: {
-          likeQuery: `%${q}%`,
+          likeQuery: '%' + q + '%',
           userId,
-          topicIds,
           limit
         }
       }
     );
 
-    const followedUsers = users.filter((item) => item.user_id_follower !== null);
-
-    const unfollowedUsers = users.filter((item) => item.user_id_follower === null);
+    let followedUsers = [];
+    let unfollowedUsers = [];
+    users.forEach((item) => {
+      if (item.user_id_follower !== null) {
+        followedUsers.push(item);
+      } else {
+        unfollowedUsers.push(item);
+      }
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The separation of followed and unfollowed users is done after the database query. This could potentially slow down the response time when dealing with large amounts of data. Consider optimizing this by separating these users during the SQL query itself, if possible.

Overall, the changes seem to improve the performance of the user search functionality. However, please consider the above points to further enhance security and efficiency.


<!-- This is an auto-generated comment by OSS CodeRabbit -->